test/IISIntegration.FunctionalTests/Utilities/RetryHandler.cs

```diff
@@ -38,7 +39,10 @@ public RetryHandler(HttpMessageHandler innerHandler, ILogger logger)
                     _logger.LogWarning("Error sending request", ex);
                 }
 
-                if (response?.IsSuccessStatusCode == true)
+
+                // Retry only on 503 that is expected during IIS startup
+                if (response != null &&
```

Comment:
HttpClient is guaranteed not to return null.

Reply:
It would be null if an exception is thrown.